test/should_fail/var_pattern_bound.erl

```diff
@@ -0,0 +1,16 @@
+-module(var_pattern_bound).
+
+-export([f/1]).
+
+-spec f(atom()) -> integer().
+f(Atom) ->
+    case g() of
+        {Atom} ->
```

Comment:
Testing `{Atom} ->` is great, but given the name of the module I'm expecting a bunch of other tests as well, e.g. `Atom ->`.

Reply:
I just wanted to add a negative test that fails without the fix. `Atom ->` already passed because `add_type_pat` for a straight var handles it without calling `add_any_types_pat`. We need a var nested in a pattern to provoke the error. `Atom ->` is already covered according to coverage but I can add one here too, should I?

Reply:
I think it would be a good idea to add a second, similar, function testing `Atom ->` unless we have a test case that explicitly tests this already.